src/plugins/auto_batch/src/sync_infer_request.cpp

```diff
@@ -5,324 +5,100 @@
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::FP16:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::FP16>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::BF16:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::BF16>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::U64:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::U64>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::I64:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::I64>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::U8:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::U8>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        case InferenceEngine::Precision::BOOL:
-            res = create_shared_blob_on_top_of_batched_blob<InferenceEngine::Precision::BOOL>(
-                m_batched_request_wrapper._inferRequestBatched->GetBlob(it.first),
-                it.first,
-                batchedInputs,
-                m_batch_id,
-                m_batch_size);
-            break;
-        default:
-            IE_THROW() << "Unsupported input precision " << it.second->getTensorDesc().getPrecision();
-        }
-        _inputs[it.first] = res;
+void SyncInferRequest::share_tensors_with_batched_req(const std::set<std::string>& batched_inputs,
+                                                      const std::set<std::string>& batched_outputs) {
+    for (const auto& it : get_inputs()) {
+        auto name = ov::op::util::get_ie_output_name(it);
+        ov::SoPtr<ov::ITensor> res;
+        auto batched_tensor = m_batched_request_wrapper->_infer_request_batched->get_tensor(it);
+        batched_tensor._so = m_batched_request_wrapper->_infer_request_batched._so;
```

Comment:
It is not correct change.
```suggestion
        if (!batched_tensor._so)
            batched_tensor._so = m_batched_request_wrapper->_infer_request_batched._so;
```